### PR TITLE
docs: guidelines on large contributions and feature branches

### PR DESCRIPTION
**Pull request by mpoke**
## Description

Closes: NA

Adds guidelines on large contributions and feature branches. Based on https://github.com/cosmos/interchain-security/pull/868 

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] included the correct `docs:` prefix in the PR title
- [x] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] Confirmed the correct `docs:` prefix in the PR title
- [ ] Confirmed all author checklist items have been addressed 
- [ ] Confirmed that this PR only changes documentation
- [ ] Reviewed content for consistency
- [ ] Reviewed content for thoroughness
- [ ] Reviewed content for spelling and grammar
- [ ] Tested instructions (if applicable)

### CODE REVIEW

**Review by glnro — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -15,9 +19,57 @@
...
 Thank you for considering making contributions to Gaia! 🎉👍
 
+## Overview
+
 Contributing to this repo can mean many things such as participating in
-discussion or proposing code changes. To ensure a smooth workflow for all
-contributors, a general procedure for contributing has been established:
+discussion or proposing code changes. 
+Following the processes outlined in this document will lead to the best 
+chance of landing changes in a release.
```
Comment (glnro) [inline CONTRIBUTING.md:27]:
```suggestion
Following the processes outlined in this document will lead to the best 
chance of getting changes merged into the codebase.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -15,9 +19,57 @@
...
+
+### Stewarding team responsibility
```
Comment (glnro) [inline CONTRIBUTING.md:29]:
```suggestion
### Responsibilities of the stewarding team
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -15,9 +19,57 @@
...
+This means that the stewarding team needs to understand the nature of, 
+and agree to maintain, all of the changes that land on `main` or a backport branch. 
+It may cost a few days/weeks' worth of time to _submit_ a particular change, 
+but _maintaining_ that change over the years has a much higher cost that the stewarding team will bear.
+
+### Ease of reviewing
+
+ The fact that the stewarding team needs to be able to deeply understand the short-,
+ medium- and long-term consequences of incoming changes means that changes need
+ to be **easily reviewed**.
```
Comment (glnro) [inline CONTRIBUTING.md:43]:
```suggestion
 to be **easy to review**.
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -15,9 +19,57 @@
...
+
+ What makes a change easy to review, and more likely to land in an upcoming
+ release?
+
+ 1. **Each pull request must do _one thing_**. It must be very clear what that
+    one thing is when looking at the pull request title, description, and linked
+    issues. It must also be very clear what value it ultimately aims to deliver,
+    and to which user(s). A single pull request that does multiple things, or
```
Comment (glnro) [inline CONTRIBUTING.md:51]:
```suggestion
    and for which user(s). A single pull request that does multiple things, or
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -32,40 +84,53 @@ contributors, a general procedure for contributing has been established:
...
-Code is merged into main through pull request procedure.
+`main` must be stable, include only completed features and never fail `make lint`, `make run-tests`, or `make build/install`.
+
+Depending on the scope of the work, we differentiate between self-contained pull requests and long-lived contributions (features).
+
+**Self-contained pull requests**:
+
+* Fork the repo (core developers must create a branch directly in the Gaia repo),
+branch from the HEAD of `main`, make some commits, and submit a PR to `main`.
+* For core developers working within the `gaia` repo, follow branch name conventions to ensure clear
```
Comment (glnro) [inline CONTRIBUTING.md:111]:
```suggestion
* For developers who are core contributors and are working within the `gaia` repo, follow branch name conventions to ensure clear
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -32,40 +84,53 @@ contributors, a general procedure for contributing has been established:
...
+ownership of branches: `{moniker}/{issue#}-branch-name`.
+* See [Branching Model](#branching-model-and-release) for more details.
+
+**Large contributions**:
+
+* Make sure that a feature branch in created in the repo.
```
Comment (glnro) [inline CONTRIBUTING.md:117]:
```suggestion
* Make sure that a feature branch is created in the repo. 
```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -190,9 +257,11 @@ Gaia follows [semantic versioning](https://semver.org), but with the following d
 
 ### PR Targeting
 
-Ensure that you base and target your PR on the `main` branch.
+Ensure that you base and target your PRs on either `main` or a feature branch.
 
-All feature additions and all bug fixes must be targeted against `main`. Exception is for bug fixes which are only related to a released version. In that case, the related bug fix PRs must target against the release branch.
+All complete features and all bug fixes must be targeted against `main`.
```
Comment (glnro) [inline CONTRIBUTING.md:262]:
```suggestion
All complete features and bug fixes must be targeted against `main`. 
```

**Review by mmulji-ic — APPROVED** [review]

**Comment by mmulji-ic** [comment]
@glnro @mpoke , sorry saw that it was approved so I merged it, didn't see the comments. @glnro please reopen/revert if need be.

**Comment by glnro** [comment]
Yeah they're just minor improvements, someone could just open a subsequent pr

**Pull request merged**